### ISSUE TITLE
scale of imputed values

### ISSUE THREAD

**Issue by Winnie09** [divyanshu-talwar/AutoImpute#5]
Hi,
It seems that the imputed matrix has both positive and negative values. Also the maximum value is larger than 300. Just wondering whether this is log transformed? If that's log transformed why there is number as large as 300? Thanks!

**Comment by divyanshu-talwar** [comment]
Hi,

The input to `AutoImpute.py` is a not a raw count matrix but a pre-processed one (which is log-transformed, among other steps). 
I suspect you skipped the pre-processing step and as a result, might be getting such large values in the output matrix. In the normalization phase of pre-processing, the input is scaled appropriately and AutoImpute being an autoencoder setup mimics the same normalization properties in its output.

To pre-process any dataset place it's raw form (in `.csv` format) in `AutoImpute\ Model/data/raw/` and run the R script `pre-process.R` resting in the directory `AutoImpute\ Model/Pre-processing/` using the following command:
> Rscript pre-process.R <input-file-name> <dataset-name>

For best results, it is advised to make sure that the hyper-parameters associated with the autoencoder model have been fine-tuned well enough by checking if the model's loss converges just fine.

Cheers!

**Comment by Winnie09** [comment]
Thanks, author for the detailed answer! I have a further question:

Does “ a pre-processed one (which is log-transformed)” means the input is processed count matrix normalized by library size and log2-transformed? My concern is that people do log2-transformed based on library size-normalized version instead of raw count. And I would like to make sure about this on autoimpute.

Thanks a lot for the help!


On Mar 20, 2019, at 11:34 AM, Divyanshu Talwar <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi,

The input to AutoImpute.py is a not a raw count matrix but a pre-processed one (which is log-transformed, among other steps).
I suspect you skipped the pre-processing step and as a result, might be getting such large values in the output matrix. In the normalization phase of pre-processing, the input is scaled appropriately and AutoImpute being an autoencoder setup mimics the same normalization properties in its output.

To pre-process any dataset place it's raw form (in .csv format) in AutoImpute\ Model/data/raw/ and run the R script pre-process.R resting in the directory AutoImpute\ Model/Pre-processing/ using the following command:

Rscript pre-process.R

For best results, it is advised to make sure that the hyper-parameters associated with the autoencoder model have been fine-tuned well enough by checking if the model's loss converges just fine.

Cheers!

**Comment by divyanshu-talwar** [comment]
Yes, the log transformation is done after the library size normalisation of the raw count matrix. For more on the steps involved in pre-processing please refer the paper or the `README`.

**Comment by Winnie09** [comment]
This is very helpful! Thanks!
On Mar 31, 2019, at 4:35 AM, Divyanshu Talwar <notifications@github.com<mailto:notifications@github.com>> wrote:


Yes, the log transformation is done after the library size normalisation of the raw count matrix. For more on the steps involved in pre-processing please refer the paper or the README.

**Comment by Winnie09** [comment]
Hi author, 

I tried what you said on a processed dataset. I used library-size-normalized and log2-transformed data as input, and found the output  contains negative values. The range of the values is from -27.74362 to 19.21706. Is there any possible cause of this problem, and do you have any suggestions to deal with it? Thanks a lot!

**Comment by chenxofhit** [comment]
@Winnie09 Does this issue still exist? I also think this issue may exist in my first glance at the architecture of this neural network.
